### PR TITLE
Colm/hts 50/intergrate with goverment gateway

### PR DESCRIPTION
**Pull request by cavster**

### CODE REVIEW

File: app/uk/gov/hmrc/helptosavefrontend/auth/HelpToSaveAuthentication.scala [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.helptosavefrontend.auth
+
+import play.api.Logger
```
Comment (sureshhmrc) [inline app/uk/gov/hmrc/helptosavefrontend/auth/HelpToSaveAuthentication.scala]: unused import ?

File: app/uk/gov/hmrc/helptosavefrontend/config/frontendAppConfig.scala [hunk 2]
```diff
@@ -33,6 +33,15 @@ object FrontendAppConfig extends AppConfig with ServicesConfig {
   private val contactHost = configuration.getString(s"contact-frontend.host").getOrElse("")
   private val contactFormServiceIdentifier = "MyService"
 
+  lazy val logInContinueURL = getConfString("auth.login-callback.base-url", "")
+  lazy val loginURL = getConfString("auth.login-callback.path", "")
+  lazy  val redirectUrlForAuth = logInContinueURL + loginURL
+
+  private lazy val companyAuthFrontend = getConfString("company-auth.url", throw new RuntimeException("Company auth url required"))
+  private lazy val companyAuthSignInPath = getConfString("company-auth.sign-in-path", "")
+  private lazy val companyAuthSignOutPath = getConfString("company-auth.sign-out-path", "")
```
Comment (sureshhmrc) [inline app/uk/gov/hmrc/helptosavefrontend/config/frontendAppConfig.scala]: `companyAuthSignOutPath` is this being used any where ?

File: app/uk/gov/hmrc/helptosavefrontend/auth/HelpToSaveAuthentication.scala [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+}
+
+case class HtsRegime(authenticationProvider: AuthenticationProvider) extends TaxRegime {
+
+  //todo find out what to do here ????
+  override def isAuthorised(accounts: Accounts): Boolean = true
+
+  override def authenticationType: AuthenticationProvider = authenticationProvider
+
+  override def unauthorisedLandingPage = {
```
Comment (sureshhmrc) [inline app/uk/gov/hmrc/helptosavefrontend/auth/HelpToSaveAuthentication.scala]: add some return type here too :)

File: app/uk/gov/hmrc/helptosavefrontend/auth/HelpToSaveAuthentication.scala [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+}
+
+case class HtsRegime(authenticationProvider: AuthenticationProvider) extends TaxRegime {
+
+  //todo find out what to do here ????
+  override def isAuthorised(accounts: Accounts): Boolean = true
+
+  override def authenticationType: AuthenticationProvider = authenticationProvider
+
+  override def unauthorisedLandingPage: Some[String] = {
```
Comment (AndyHWChung) [inline app/uk/gov/hmrc/helptosavefrontend/auth/HelpToSaveAuthentication.scala]: Option[String]?

File: app/uk/gov/hmrc/helptosavefrontend/config/frontendAppConfig.scala [hunk 5]
```diff
@@ -33,6 +33,15 @@ object FrontendAppConfig extends AppConfig with ServicesConfig {
   private val contactHost = configuration.getString(s"contact-frontend.host").getOrElse("")
   private val contactFormServiceIdentifier = "MyService"
 
+  lazy val logInContinueURL = getConfString("auth.login-callback.base-url", "")
```
Comment (AndyHWChung) [inline app/uk/gov/hmrc/helptosavefrontend/config/frontendAppConfig.scala]: why are these lazy?

**Pull request merged**